include/fmt/std.h

```diff
@@ -713,9 +713,27 @@ template <typename T, typename Char> struct formatter<std::complex<T>, Char> {
   }
 };
 
+template <typename T, typename Enable = void>
+struct has_format_as : std::false_type {};
+template <typename T>
+struct has_format_as<T, void_t<decltype(format_as(std::declval<const T&>()))>>
+    : std::true_type {};
+
+template <typename T, typename Enable = void>
+struct has_format_as_member : std::false_type {};
+template <typename T>
+struct has_format_as_member<
+    T, void_t<decltype(formatter<T>::format_as(std::declval<const T&>()))>>
+    : std::true_type {};
+
+// Guard against format_as because reference_wrappers are implicitly convertible
+// to T&
+
 template <typename T, typename Char>
-struct formatter<std::reference_wrapper<T>, Char,
-                 enable_if_t<is_formattable<remove_cvref_t<T>, Char>::value>>
+struct formatter<
+    std::reference_wrapper<T>, Char,
+    enable_if_t<is_formattable<remove_cvref_t<T>, Char>::value &&
+                !has_format_as<T>::value && !has_format_as_member<T>::value>>
```

Comment:
Is it possible to reuse ``use_format_as`` and ``use_format_as_member``?

https://github.com/fmtlib/fmt/blob/59259a5fde0d555da6faa1f300e83cc2eba61193/include/fmt/base.h#L1120-L1141

Reply:
Unfortunately this doesn't work because of the `std::is_arithmetic`. Consider, e.g.,
```cpp
struct type_with_format_as_string {
    std::string str;
};

std::string format_as(const type_with_format_as_string& t) { return t.str; }
```

`use_format_as<type_with_format_as_string >::value` is false however `fmt::format("{}", type_with_format_as_string{"foo"})` works as it should. As such, attempting to use `use_format_as` for the check results in the same ambiguities.